File: novel-backend/novel-service/src/main/java/com/novel/controller/front/CommentController.java
package com.novel.controller.front;

import cn.dev33.satoken.annotation.SaCheckLogin;
import com.novel.result.PageResult;
import com.novel.result.Result;
import com.novel.service.BookCommentService;
import com.novel.user.dto.comment.BookCommentInput;
import com.novel.user.dto.comment.BookCommentView;
import com.novel.user.dto.comment.BookSubCommentView;
import com.novel.utils.IpUtils;
import io.swagger.v3.oas.annotations.Operation;
import io.swagger.v3.oas.annotations.tags.Tag;
import jakarta.servlet.http.HttpServletRequest;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@Tag(name = "评论")
@RestController
@RequestMapping("/comment")
public class CommentController {
	
	@Autowired
	private BookCommentService bookCommentService;
	
	/**
	 * 小说评论查询接口 顶层评论
	 * @param bookId
	 * @return
	 */
	@Operation(summary = "小说根评论")
	@GetMapping("/top")
	public Result<PageResult<BookCommentView>> listBookComments(
			long bookId,
			@RequestParam(defaultValue = "1") Integer pageNo,
			@RequestParam(defaultValue = "20") Integer pageSize) {
		PageResult<BookCommentView> pageResult = bookCommentService.listTopBookComments(bookId, pageNo, pageSize);
		return Result.ok(pageResult);
	}
	
	@Operation(summary = "小说子评论")
	@GetMapping("/child")
	public Result<List<BookSubCommentView>> listBookComments(
			long bookId,
			long rootId) {
		List<BookSubCommentView> list = bookCommentService.listChildBookComments(bookId, rootId);
		return Result.ok(list);
	}
	
	@SaCheckLogin
	@Operation(summary = "发布评论")
	@PostMapping("/add")
	public Result<BookCommentView> addBookComment(HttpServletRequest request, @RequestBody BookCommentInput bookComment) {
		String ip = IpUtils.getIp(request);
		String address = IpUtils.getIpAddress(ip);
		BookCommentView bookCommentView = bookCommentService.addBookComment(ip, address, bookComment);
		return Result.ok(bookCommentView);
	}
	
	@SaCheckLogin
	@Operation(summary = "删除评论")
	@DeleteMapping("{id}")
	public Result<String> deleteBookComment(@PathVariable Long id) {
		boolean isDeleted = bookCommentService.deleteBookComment(id);
		return isDeleted ? Result.ok() : Result.fail("删除失败：当前用户无操作权限");
	}
}

File: novel-admin-react/src/__generated/services/CommentController.ts
import type {Executor} from '../';
import type {
    BookCommentInput, 
    BookCommentView, 
    BookSubCommentView, 
    PageResult, 
    Result
} from '../model/static/';

export class CommentController {
    
    constructor(private executor: Executor) {}
    
    readonly addBookComment: (options: CommentControllerOptions['addBookComment']) => Promise<
        Result<BookCommentView>
    > = async(options) => {
        let _uri = '/comment/add';
        return (await this.executor({uri: _uri, method: 'POST', body: options.body})) as Promise<Result<BookCommentView>>;
    }
    
    readonly deleteBookComment: (options: CommentControllerOptions['deleteBookComment']) => Promise<
        Result<string>
    > = async(options) => {
        let _uri = '/comment/';
        _uri += encodeURIComponent(options.id);
        return (await this.executor({uri: _uri, method: 'DELETE'})) as Promise<Result<string>>;
    }
    
    /**
     * 小说评论查询接口 顶层评论
     * @parameter {CommentControllerOptions['listBookComments']} options
     * - bookId 小说ID
     * @return 评论列表
     */
    readonly listBookComments: (options: CommentControllerOptions['listBookComments']) => Promise<
        Result<PageResult<BookCommentView>>
    > = async(options) => {
        let _uri = '/comment/top';
        let _separator = _uri.indexOf('?') === -1 ? '?' : '&';
        let _value: any = undefined;
        _value = options.bookId;
        _uri += _separator
        _uri += 'bookId='
        _uri += encodeURIComponent(_value);
        _separator = '&';
        _value = options.pageNo;
        if (_value !== undefined && _value !== null) {
            _uri += _separator
            _uri += 'pageNo='
            _uri += encodeURIComponent(_value);
            _separator = '&';
        }
        _value = options.pageSize;
        if (_value !== undefined && _value !== null) {
            _uri += _separator
            _uri += 'pageSize='
            _uri += encodeURIComponent(_value);
            _separator = '&';
        }
        return (await this.executor({uri: _uri, method: 'GET'})) as Promise<Result<PageResult<BookCommentView>>>;
    }
    
    readonly listBookComments_2: (options: CommentControllerOptions['listBookComments_2']) => Promise<
        Result<ReadonlyArray<BookSubCommentView>>
    > = async(options) => {
        let _uri = '/comment/child';
        let _separator = _uri.indexOf('?') === -1 ? '?' : '&';
        let _value: any = undefined;
        _value = options.bookId;
        _uri += _separator
        _uri += 'bookId='
        _uri += encodeURIComponent(_value);
        _separator = '&';
        _value = options.rootId;
        _uri += _separator
        _uri += 'rootId='
        _uri += encodeURIComponent(_value);
        _separator = '&';
        return (await this.executor({uri: _uri, method: 'GET'})) as Promise<Result<ReadonlyArray<BookSubCommentView>>>;
    }
}

export type CommentControllerOptions = {
    'listBookComments': {
        /**
         * 小说ID
         */
        readonly bookId: number, 
        readonly pageNo?: number | undefined, 
        readonly pageSize?: number | undefined
    }, 
    'listBookComments_2': {
        readonly bookId: number, 
        readonly rootId: number
    }, 
    'addBookComment': {
        readonly body: BookCommentInput
    }, 
    'deleteBookComment': {
        readonly id: number
    }
}
